src/utility/hexgrid.js

```diff
@@ -1215,43 +1153,87 @@ export class HexGrid {
 	}
 
 	selectHexUp() {
-		if (this.hexExists(this.selectedHex.y - 1, this.selectedHex.x)) {
-			let hex = this.hexes[this.selectedHex.y - 1][this.selectedHex.x];
-			this.selectedHex = hex;
-			hex.onSelectFn();
+		if (!this.hexExists(this.selectedHex.y - 1, this.selectedHex.x)) {
+			return;
+		}
+
+		if (this.selectedHex) {
+			this.clearHexViewAlterations();
+			this.selectedHex.onHoverOffFn(this.selectedHex);
 		}
+
+		const hex = this.hexes[this.selectedHex.y - 1][this.selectedHex.x];
+		this.selectedHex = hex;
+		hex.onSelectFn(hex);
 	}
 
 	selectHexDown() {
-		if (this.hexExists(this.selectedHex.y + 1, this.selectedHex.x)) {
-			let hex = this.hexes[this.selectedHex.y + 1][this.selectedHex.x];
-			this.selectedHex = hex;
-			hex.onSelectFn();
+		if (!this.hexExists(this.selectedHex.y + 1, this.selectedHex.x)) {
+			return;
+		}
+
+		if (this.selectedHex) {
+			this.clearHexViewAlterations();
+			this.selectedHex.onHoverOffFn(this.selectedHex);
 		}
+
+		const hex = this.hexes[this.selectedHex.y + 1][this.selectedHex.x];
+		this.selectedHex = hex;
+		hex.onSelectFn(hex);
 	}
 
 	selectHexLeft() {
-		if (this.hexExists(this.selectedHex.y, this.selectedHex.x - 1)) {
-			let hex = this.hexes[this.selectedHex.y][this.selectedHex.x - 1];
-			this.selectedHex = hex;
-			hex.onSelectFn();
+		if (!this.hexExists(this.selectedHex.y, this.selectedHex.x - 1)) {
+			return;
+		}
+
+		if (this.selectedHex) {
+			this.clearHexViewAlterations();
+			this.selectedHex.onHoverOffFn(this.selectedHex);
 		}
+
+		const hex = this.hexes[this.selectedHex.y][this.selectedHex.x - 1];
+		this.selectedHex = hex;
+		hex.onSelectFn(hex);
 	}
 
 	selectHexRight() {
-		if (this.hexExists(this.selectedHex.y, this.selectedHex.x + 1)) {
-			let hex = this.hexes[this.selectedHex.y][this.selectedHex.x + 1];
-			this.selectedHex = hex;
-			hex.onSelectFn();
+		if (!this.hexExists(this.selectedHex.y, this.selectedHex.x + 1)) {
+			return;
 		}
+
+		if (this.selectedHex) {
+			this.clearHexViewAlterations();
+			this.selectedHex.onHoverOffFn(this.selectedHex);
+		}
+
+		const hex = this.hexes[this.selectedHex.y][this.selectedHex.x + 1];
+		this.selectedHex = hex;
+		hex.onSelectFn(hex);
 	}
 
-	confirmHex(hex) {
+	confirmHex() {
 		if (this.game.freezedInput) {
 			return;
 		}
 
-		this.selectedHex.onConfirmFn(hex);
+		this.selectedHex.onConfirmFn(this.selectedHex);
+	}
+
+	/**
+	 * Reset the visual state for hexes that might have been hovered, dashed, etc.
+	 * Note: I'm not entirely sure what this code is doing.
+	 */
+	clearHexViewAlterations() {
+		if (!this.selectedHex) {
+			return;
+		}
+
+		this.redoLastQuery();
```

Comment:
I wasn't able to completely understand this code, but it was moved as-is from Hex "mouseleave".